superset/security.py

```diff
@@ -30,11 +31,17 @@
 )
 from flask_appbuilder.widgets import ListWidget
 from sqlalchemy import or_
+from sqlalchemy.engine.base import Connection
+from sqlalchemy.orm.mapper import Mapper
 
 from superset import sql_parse
 from superset.connectors.connector_registry import ConnectorRegistry
 from superset.exceptions import SupersetSecurityException
-from superset.utils.core import DatasourceName
+
+if TYPE_CHECKING:
+    from superset.models.core import Database, BaseDatasource
+
+from superset.utils.core import DatasourceName  # noqa: I202
```

Comment:
Perhaps we should add ignoring of `I202` to `tox.ini`? @sturmer ran intro a few of those in #7643 and it seems `flake8` and `black` aren't aligned in their thinking on newlines.

Reply:
@villebro I can create another PR which ignores `I202` and removes all references.